include/circt/Dialect/FIRRTL/FIRRTLCanonicalization.td

```diff
@@ -134,5 +146,57 @@ def SubWithInvalidOp : Pat<
     (KnownWidth $x), (IsInvalid $y)
   ]>;
 
+////////////////////////////////////////////////////////////////////////////////
+// DontTouch application
+////////////////////////////////////////////////////////////////////////////////
+
+// node(x,name),!dnt -> node(x, "")
+def DropNameNode : Pat<
+  (NodeOp $x, $name, $annotations),
+  (NodeOp $x, (GetEmptyString), $annotations),
+  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+
+// wire(name),!dnt -> wire("")
+def DropNameWire : Pat<
+  (WireOp $name, $annotations),
+  (WireOp (GetEmptyString), $annotations),
+  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+
+// reg(clk, name),!dnt -> reg(clk, "")
+def DropNameReg : Pat<
+  (RegOp $clk, $name, $annotations),
+  (RegOp $clk, (GetEmptyString), $annotations),
+  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+
+// regreset(clk, sig, val, name),!dnt -> regreset(clk, sig, val, "")
+def DropNameRegReset : Pat<
+  (RegResetOp $clk, $sig, $val, $name, $annotations),
+  (RegResetOp $clk, $sig, $val, (GetEmptyString), $annotations),
+  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+
+// Mem(...,name),!dnt -> Mem(...x, "")
+def DropNameMem : Pat<
+  (MemOp $rd, $rw, $depth, $ruw, $portNames, $name, $annotations, $portAnno),
+  (MemOp $rd, $rw, $depth, $ruw, $portNames, (GetEmptyString), $annotations, $portAnno),
+  [(uselessNameAttr $name), (NoDontTouch $annotations), (NoPortsDontTouch $portAnno)]>;
+
+// CombMem(name),!dnt -> CombMem("")
+def DropNameCombMem : Pat<
+  (CombMemOp $name, $annotations),
+  (CombMemOp (GetEmptyString), $annotations),
+  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+
+// SeqMem(ruw, name),!dnt -> SeqMem(ruw, "")
+def DropNameSeqMem : Pat<
+  (SeqMemOp $ruw, $name, $annotations),
+  (SeqMemOp $ruw, (GetEmptyString), $annotations),
+  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
+
+// memoryport(..., name),!dnt -> memoryport(..., "")
+def DropNameMemoryPort : Pat<
+  (MemoryPortOp  $memory, $direction, $name, $annotations),
+  (MemoryPortOp  $memory, $direction, (GetEmptyString), $annotations),
+  [(uselessNameAttr $name), (NoDontTouch $annotations)]>;
```

Comment:
These look great! I wish there was some way to simplify these into a common class, but I'm not seeing an obvious way to do this.  (No need to try to come up with one....)

Reply:
It would also be nice to have the patterns let you now specify stuff without dropping them.  Here is the field I care about, keep all the other fields the same in the clone.